src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitWorkerService.java

```diff
@@ -0,0 +1,240 @@
+        if (otherNetworkUuids.isEmpty()) {
+            return network;
+        } else {
+            List<Network> otherNetworks = otherNetworkUuids.stream().map(uuid -> getNetwork(uuid, variantId)).collect(Collectors.toList());
+            List<Network> networks = new ArrayList<>();
+            networks.add(network);
+            networks.addAll(otherNetworks);
+            MergingView mergingView = MergingView.create("merge", "iidm");
+            mergingView.merge(networks.toArray(new Network[0]));
+            return mergingView;
+        }
+    }
+
+    public ShortCircuitAnalysisResult run(ShortCircuitRunContext context) {
+        try {
+            return run(context, null);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            return null;
+        } catch (Exception e) {
+            LOGGER.error(FAIL_MESSAGE, e);
+            return null;
+        }
+    }
+
+    private ShortCircuitAnalysisResult run(ShortCircuitRunContext context, UUID resultUuid) throws ExecutionException, InterruptedException {
+        Objects.requireNonNull(context);
+
+        LOGGER.info("Run short circuit analysis...");
+        Network network = getNetwork(context.getNetworkUuid(), context.getOtherNetworkUuids(), context.getVariantId());
+
+        Reporter reporter = context.getReportUuid() != null ? new ReporterModel("ShortCircuitAnalysis", "Short circuit analysis") : Reporter.NO_OP;
+
+        CompletableFuture<ShortCircuitAnalysisResult> future = runShortCircuitAnalysisAsync(context, network, reporter, resultUuid);
+
+        ShortCircuitAnalysisResult result = future == null ? null : future.get();
+        if (context.getReportUuid() != null) {
+            reportService.sendReport(context.getReportUuid(), reporter);
+        }
+        return result;
+    }
+
+    private CompletableFuture<ShortCircuitAnalysisResult> runShortCircuitAnalysisAsync(ShortCircuitRunContext context,
+                                                                                       Network network,
+                                                                                       Reporter reporter,
+                                                                                       UUID resultUuid) {
+        lockRunAndCancelShortCircuitAnalysis.lock();
+        try {
+            if (resultUuid != null && cancelComputationRequests.get(resultUuid) != null) {
+                return null;
+            }
+            //TODO set provider properly
+            ShortCircuitAnalysis.Runner shortCircuitAnalysisRunner = shortCircuitAnalysisFactorySupplier.apply("Courcirc");
+            network.getVariantManager().setWorkingVariant(context.getVariantId() != null ? context.getVariantId() : VariantManagerConstants.INITIAL_VARIANT_ID);
+
+            CompletableFuture<ShortCircuitAnalysisResult> future = shortCircuitAnalysisRunner.runAsync(
+                network,
+                List.of(),
```

Comment:
Faults in ShortCircuitRunContext are not given as parameter to the runAsync method

Reply:
For this first version, we want to do a systemactic analysis. To do so, faults on all buses have to be created and passed to the model. I added the faults' creation.

Reply:
Finally I removed Faults parameter from the run-and-save endpoint.